### PR TITLE
Nested Types

### PR DESCRIPTION
**Pull request by BigEpsilon**
This PR adds the following:
1- Solves #105
2- Add support for parsing "typedef typename" definitions.
3- Maps nested structs and classes to their correct cpp structs/classes and add the parrent struct/classe to their name (see new test for better explanation).

### CODE REVIEW

**Comment by BigEpsilon** [comment]
These test fails are strange as all tests pass locally. And the fails do not seem to be related to this PR.
Any way I still need to do some work on methods, constructors and destructors so they are translated correctly.

**Comment by BigEpsilon** [comment]
So I think this PR is ready.
Additionally to what the first point, this PR also fixes #111.
More work have to be done in order to support constructors and destructors for classes with more than one generic parameters, but I think this PR is already too heavy, so better let that for another PR.

File: testsuite/results/nested.nim [hunk 1]
```diff
@@ -0,0 +1,88 @@
+nil
```
Comment (Araq) [inline testsuite/results/nested.nim:1]: Wait what? This cannot be correct... :-)

**Comment by Araq** [comment]
Sorry this took so long... :-/

**Pull request merged**